database-commons/src/main/java/io/cdap/plugin/db/batch/sink/AbstractDBSink.java

```diff
@@ -189,18 +197,33 @@ private Schema inferSchema(Class<? extends Driver> driverClass) {
                                           dbSinkConfig.getJdbcPluginName());
       Properties connectionProperties = new Properties();
       connectionProperties.putAll(dbSinkConfig.getConnectionArguments());
-      try (Connection connection = DriverManager.getConnection(dbSinkConfig.getConnectionString(),
-                                                               connectionProperties)) {
-        executeInitQueries(connection, dbSinkConfig.getInitQueries());
-
-        try (Statement statement = connection.createStatement();
-             ResultSet rs = statement.executeQuery("SELECT * FROM " + dbSinkConfig.getEscapedTableName()
-                                                     + " WHERE 1 = 0")) {
-          inferredFields.addAll(getSchemaReader().getSchemaFields(rs));
+      if (dbSinkConfig.dbSchemaName == null) {
```

Comment:
similarly here you dont need big if-else blocks. You can construct `fullyQualifiedTableName` once using ternary operator and use it everywhere below while forming sql queries.